subsys/bluetooth/host/audio/iso.c

```diff
@@ -1044,15 +1054,6 @@ void bt_iso_recv(struct bt_conn *conn, struct net_buf *buf, uint8_t flags)
 		conn->rx_len -= buf->len;
 		net_buf_unref(buf);
 
-		if (conn->rx_len) {
-			BT_ERR("Unexpected ISO end fragment");
-			return;
-		}
```

Comment:
Shouldn't we keep this handling? If the END has not finished reassembling the entire ISO frame.

Reply:
This handling was moved to lines 1040-1045. The only difference is that it checks for `!conn->rx` instead of the len (for consistency reasons).

Reply:
That check is making sure that we have started on a frame, but checking the length would be to make sure that we did in fact get them all. The rx_len should have reached zero here.